addons/website_sale/controllers/main.py

```diff
@@ -1064,3 +1064,7 @@ def country_infos(self, country, mode, **kw):
             states=[(st.id, st.name, st.code) for st in country.get_website_sale_states(mode=mode)],
             phone_code=country.phone_code
         )
+
+    @http.route(['/shop/update_carrier'], type='json', auth='public', methods=['POST'], website=True, csrf=False)
+    def update_eshop_carrier(self, **post):
+        pass
```

Comment:
This should become the only `/shop/update_carrier` route. You could have something like:
```
res1 = self._update_website_sale_delivery(...)
res2 = self._update_website_sale_coupon(...)
merge and return results
```

This way, you explicitly control which method is called when, and you don't rely on the installation order.